src/HttpHeader.cc

```diff
@@ -747,6 +748,19 @@ HttpHeader::refreshMask()
     }
 }
 
+size_t
+HttpHeader::maxLen() const
+{
+    switch (owner) {
+    case hoRequest:
+        return Config.maxRequestHeaderSize;
+    case hoReply:
+        return Config.maxReplyHeaderSize;
+    default:
+        return UnrestrictedSize;
```

Comment:
Please (re)move `UnrestrictedSize` global (here) and just return a proper type-based maximum. The caller would not have to worry about this special case then -- all sizes are effectively restricted, if not by configuration, then by our type limitations.

Reply:
I simplified and eliminated this method.